ddl/column.go

```diff
@@ -1102,6 +1144,9 @@ func (w *updateColumnWorker) fetchRowColVals(txn kv.Transaction, taskRange reorg
 			}
 
 			if err1 := w.getRowRecord(handle, recordKey, rawRow); err1 != nil {
+				if types.ErrTruncated.Equal(err1) {
+					err1 = types.ErrTruncated.GenWithStackByArgs(w.oldColInfo.Name, rowIdx)
+				}
```

Comment:
the original source of this error is coming from the cast function, you can judge this error in here.
refer my pr: https://github.com/pingcap/tidb/pull/20012